### PR TITLE
Intl: Add footer affiliate logo.

### PR DESCRIPTION
**Pull request by sergiitk**
#### What's this PR do?
- Adds paraneue_dosomething [theme setting](http://localhost:8888/admin/appearance/settings/paraneue_dosomething) to setup an affiliate logo in the footer
- Creates preprocess variables for `footer.tpl.php`
#### Where should the reviewer start?
- `drush cc all`
- Open paraneue_dosomething [theme setting](http://localhost:8888/admin/appearance/settings/paraneue_dosomething) 
#### How should this be manually tested?

footer-logo
- Add "Do Something Congo is a programme of:" to AFFILIATE LOGO → Text
- Upload attached **test logo** to AFFILIATE LOGO → File
- Save form
#### Any background context you want to provide?
- [Mockup pdf](https://docs.google.com/a/dosomething.org/viewer?a=v&pid=sites&srcid=ZG9zb21ldGhpbmcub3JnfGRvc29tZXRoaW5ndXh3aWtpfGd4OmIzMDM1MTM2MDQ5ODRkNw)
- Snippet for frontends, `footer.tpl.php`:

``` php
<?php if (isset($affiliate_logo)): ?>
<div class="affiliate-logo">
  <p><?php print $affiliate_logo['text'] ?></p>
  <?php print theme_image(array(
    'path'  => $affiliate_logo['file'],
    'title' => $affiliate_logo['text'],
    'alt'   => $affiliate_logo['text'],
  )); ?>
</div>
<?php endif; ?>
```
#### What are the relevant tickets?

Closes [#DOS-19](https://jira.dosomething.org/browse/DOS-19)

### CODE REVIEW

**Comment by sergiitk** [comment]
@aaronschachter Could you please take a look? @tjrosario will be implementing frontend

File: lib/themes/dosomething/paraneue_dosomething/theme-settings.php [hunk 1]
```diff
@@ -108,17 +108,50 @@ function _paraneue_dosomething_theme_settings_footer(&$form, $form_state) {
     '#type' => 'fieldset',
     '#title' => t('Footer'),
   );
+  $footer = &$form['footer'];
```
Comment (aaronschachter) [inline lib/themes/dosomething/paraneue_dosomething/theme-settings.php:164]: Is this intentionally `&$form` and not `$form` ?

File: lib/themes/dosomething/paraneue_dosomething/theme-settings.php [hunk 2]
```diff
@@ -132,8 +165,8 @@ function _paraneue_dosomething_theme_settings_footer(&$form, $form_state) {
     '#default_value' => theme_get_setting('footer_social')
   );
 
+  $links = &$form['links'];
```
Comment (aaronschachter) [inline lib/themes/dosomething/paraneue_dosomething/theme-settings.php:221]: Same question here

**Comment by aaronschachter** [comment]
Do we have to set the file status to 1 after saving, in order to prevent it from being deleted upon cron?  Same goes for #2824. I used a `managed_file` upload for the User Edit form in #2832, and realized that it was saved with status 0.

Assuming this is the case, I've reopened https://jira.dosomething.org/browse/DOS-58

https://api.drupal.org/api/drupal/modules!system!system.module/function/system_cron/7

File: lib/themes/dosomething/paraneue_dosomething/theme-settings.php (hunk 2 above)
Comment (aaronschachter) [inline lib/themes/dosomething/paraneue_dosomething/theme-settings.php:221]: Should this not just be `$form['links']` ?

File: lib/themes/dosomething/paraneue_dosomething/theme-settings.php [hunk 3]
```diff
@@ -57,6 +57,43 @@ function paraneue_dosomething_form_system_theme_settings_alter(&$form, $form_sta
...
+    return;
+  }
+  $file = file_load($input['footer_affiliate_logo_file']['fid']);
+  if (empty($file)) {
+    return;
+  }
+  if (!empty($input['footer_affiliate_logo'])) {
+    $changed      = $file->status != FILE_STATUS_PERMANENT;
+    $file->status = FILE_STATUS_PERMANENT;
+  } else {
```
Comment (aaronschachter) [inline lib/themes/dosomething/paraneue_dosomething/theme-settings.php]: A little picky, but isn't best practice to keep `else` statements in a separate line? https://www.drupal.org/coding-standards#controlstruct

Comment (sergiitk) [inline lib/themes/dosomething/paraneue_dosomething/theme-settings.php]: No, you're right. It's just me used to it. Will fix that.

File: lib/themes/dosomething/paraneue_dosomething/theme-settings.php [hunk 4]
```diff
@@ -57,6 +57,43 @@ function paraneue_dosomething_form_system_theme_settings_alter(&$form, $form_sta
...
+    $changed      = $file->status != 0;
```
Comment (aaronschachter) [inline lib/themes/dosomething/paraneue_dosomething/theme-settings.php:96]: I don't totally get what's going on here... maybe comments to explain?

File: lib/themes/dosomething/paraneue_dosomething/theme-settings.php [hunk 5]
```diff
@@ -57,6 +57,43 @@ function paraneue_dosomething_form_system_theme_settings_alter(&$form, $form_sta
...
+    $file->status = 0;
+  }
+  if ($changed) {
+    file_save($file);
+    if ($file->status == FILE_STATUS_PERMANENT) {
+      file_usage_add($file, 'dosomething_settings', 'dosomething_settings', $file->fid);
+    } else {
```
Comment (aaronschachter) [inline lib/themes/dosomething/paraneue_dosomething/theme-settings.php]: Same re: new line

File: lib/themes/dosomething/paraneue_dosomething/theme-settings.php (hunk 1 above)
Comment (sergiitk) [inline lib/themes/dosomething/paraneue_dosomething/theme-settings.php:164]: Yes, that's just a shortcut. BTW, it use to be like that, see removed line 118.

File: lib/themes/dosomething/paraneue_dosomething/theme-settings.php [hunk 6]
```diff
@@ -57,6 +57,43 @@ function paraneue_dosomething_form_system_theme_settings_alter(&$form, $form_sta
...
+    $changed      = $file->status != FILE_STATUS_PERMANENT;
+    $file->status = FILE_STATUS_PERMANENT;
+  } else {
+    $changed      = $file->status != 0;
+    $file->status = 0;
+  }
+  if ($changed) {
+    file_save($file);
+    if ($file->status == FILE_STATUS_PERMANENT) {
+      file_usage_add($file, 'dosomething_settings', 'dosomething_settings', $file->fid);
```
Comment (aaronschachter) [inline lib/themes/dosomething/paraneue_dosomething/theme-settings.php]: Does it make sense to change the usage to `paraneue_dosomething`? Technically that's what's using the file... `dosomething_settings` module is not using it all..

File: lib/themes/dosomething/paraneue_dosomething/theme-settings.php (hunk 4 above)
Comment (sergiitk) [inline lib/themes/dosomething/paraneue_dosomething/theme-settings.php:96]: Agree, will throw in some comments.

File: lib/themes/dosomething/paraneue_dosomething/theme-settings.php (hunk 6 above)
Comment (sergiitk) [inline lib/themes/dosomething/paraneue_dosomething/theme-settings.php]: Yes, I agree. As an alternative I considered `system`, but eventually stopped on `dosomething_settings`. What would be your choice here?

File: lib/themes/dosomething/paraneue_dosomething/theme-settings.php (hunk 2 above)
Comment (sergiitk) [inline lib/themes/dosomething/paraneue_dosomething/theme-settings.php:221]: No, same here. Just a shortcut. Moved from line 120.

File: lib/themes/dosomething/paraneue_dosomething/theme-settings.php (hunk 6 above)
Comment (aaronschachter) [inline lib/themes/dosomething/paraneue_dosomething/theme-settings.php]: I personally think it's fine to just use `paraneue_dosomething` as the module name even though its technically not a module....  what do you think?

Comment (sergiitk) [inline lib/themes/dosomething/paraneue_dosomething/theme-settings.php]: Yep, thought about it also. Let me check if it saves the reference this way.

**Comment by aaronschachter** [comment]
Looks good :+1:  ship it

**Comment by sergiitk** [comment]
@tjrosario, it's finally merged! You can start the theming part when you're ready.

**Pull request merged**